### PR TITLE
Draft: Improvements

### PR DESCRIPTION
**Pull request by panther7**
see commits

### CODE REVIEW

**Comment by panther7** [comment]
@laedit you could append some commits and improving more things...

**Comment by laedit** [comment]
Thanks, I will look into it.
Do you have suggestions for other improvements?

File: src/options.ts [hunk 1]
```diff
@@ -1,9 +1,28 @@
...
+  throw new Error('Invalid input element');
 }
 
-function setInputValue(elementId: string, value: string): void {
-  (document.getElementById(elementId) as HTMLInputElement).value = value;
+function setInputValue(elementId: string, value: string | number | boolean): void {
+  const elm = document.getElementById(elementId)! as HTMLInputElement;
+  if (elm) {
+    if (elm.type === "checkbox") {
+      elm.checked = !!value;
```
Comment (laedit) [inline src/options.ts:20]: I am curious, why the use of a double `!`?

Comment (panther7) [reply src/options.ts:20]: Is it alias for `Boolean(value)`, this is using, if you want change data type to bool.

**Comment by panther7** [comment]
More improvements posible:
* rewrite settings to native firefox UI (if is it possible :))
* add "settings button" into "new tab" page, maybe right/bottom right
* posible set backgroud as image (via settings UI)

**Comment by laedit** [comment]
Thanks for the suggestions!
To my knowledge, it is not possible to write options in native UI, but I will keep the other two suggestions for later :)

File: src/settings.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const defaultSettings: Settings = {
-    background: "#FAFAFA",
+    background: "#fafafa",
     foreground: "#424242",
+    colorScheme: "default",
```
Comment (laedit) [inline src/settings.ts:4]:
```suggestion
    colorScheme: "custom",
```
This should be `custom` to handle already existing custom colors.

File: src/settings.ts [hunk 3]
```diff
@@ -16,10 +17,12 @@ const defaultSettings: Settings = {
 type TempUnit = "celsius" | "fahrenheit";
 type ClockType = "12" | "24";
 type FontType = "thin" | "bold";
+type ColorScheme = "default" | "light" | "dark" | "custom";
```
Comment (laedit) [inline src/settings.ts:20]: I think that `system` instead of `default` will better indicate what it represents.

**Review by laedit — CHANGES_REQUESTED** [review]
Sorry for my late response, I haven't got the time last weekend.
There are just a couple of changes and this will be merged.
Thanks again for your contribution!

**Pull request merged**